lib/ansible/playbook/play_context.py

```diff
@@ -446,130 +380,30 @@ def set_task_and_variable_override(self, task, variables, templar):
 
     def make_become_cmd(self, cmd, executable=None):
         """ helper function to create privilege escalation commands """
+        # DEPRECATED
 
-        prompt = None
-        success_key = None
-        self.prompt = None
+        if cmd and self.become:
```

Comment:
I'd flip this and return early and deindent the rest of the method

(completely untested example)

``` python
   def make_become_cmd(self, cmd, executable=None):
        if not cmd or not self.become:
            return cmd

        # load/call become plugins here
        plugin = become_loader.get(self.become_method)
        if not plugin:
            raise AnsibleError("Privilege escalation method not found: %s" % self.become_method)

        options = {'become_exe': self.become_exe or getattr(self, '%s_exe' % self.become_method, self.become_method) or self.become_method,
                   'become_flags': self.become_flags or getattr(self, '%s_flags' % self.become_method, '') or '',
                   'become_user': self.become_user,
                   'become_pass': self.become_pass}
        plugin.set_options(direct=options)

        if not executable:
            executable = self.executable

        shell = get_shell_plugin(executable=executable)
        cmd = plugin.build_become_command(cmd, shell)

        # for backwards compat:
        self.prompt = plugin.prompt

        return cmd
```

Reply:
i prefer to keep 1 return, it seems to loose a lot of people when you have more than one exist from the function

Reply:
@alikins +1  That is better coding style in this case.   If we have a short bit of logic that removes certain specific cases from the rest of the code, it makes sense to return early.  That way we don't have to wonder throughout the rest of the function what happens to the other cases.  https://youtu.be/rrBJVMyD-Gs?t=755